web/packages/teleterm/src/services/tshd/createClient.ts

```diff
@@ -270,801 +176,475 @@ export function createTshdClient(
-            rootClusterUri: uri,
-          }),
-          err => {
-            if (err) {
-              reject(err);
-            } else {
-              resolve();
-            }
-          }
-        );
+    async deleteConnectMyComputerNode(uri: uri.RootClusterUri) {
+      await tshd.deleteConnectMyComputerNode({
+        rootClusterUri: uri,
       });
     },
 
-    getConnectMyComputerNodeName(uri: uri.RootClusterUri) {
-      return new Promise<string>((resolve, reject) => {
-        tshd.getConnectMyComputerNodeName(
-          api.GetConnectMyComputerNodeNameRequest.create({
-            rootClusterUri: uri,
-          }),
-          (err, response) => {
-            if (err) {
-              reject(err);
-            } else {
-              resolve(response.name as uri.ServerUri);
-            }
-          }
-        );
+    async getConnectMyComputerNodeName(uri: uri.RootClusterUri) {
+      const { response } = await tshd.getConnectMyComputerNodeName({
+        rootClusterUri: uri,
       });
+
+      return response.name as uri.ServerUri;
     },
 
-    updateHeadlessAuthenticationState(
+    async updateHeadlessAuthenticationState(
       params: UpdateHeadlessAuthenticationStateParams,
-      abortSignal?: types.TshAbortSignal
+      abortSignal?: CloneableAbortSignal
     ) {
-      return withAbort(abortSignal, callRef => {
-        const req = api.UpdateHeadlessAuthenticationStateRequest.create({
-          rootClusterUri: params.rootClusterUri,
-          headlessAuthenticationId: params.headlessAuthenticationId,
-          state: params.state,
-        });
-
-        return new Promise<void>((resolve, reject) => {
-          callRef.current = tshd.updateHeadlessAuthenticationState(req, err => {
-            if (err) {
-              reject(err);
-            } else {
-              resolve();
-            }
-          });
-        });
+      await tshd.updateHeadlessAuthenticationState(params, {
+        abort: abortSignal,
       });
     },
 
-    listUnifiedResources(
+    async listUnifiedResources(
       params: types.ListUnifiedResourcesRequest,
-      abortSignal?: types.TshAbortSignal
+      abortSignal?: CloneableAbortSignal
     ) {
-      return withAbort(abortSignal, callRef => {
-        const req = api.ListUnifiedResourcesRequest.create({
+      const { response } = await tshd.listUnifiedResources(
+        {
           clusterUri: params.clusterUri,
           limit: params.limit,
           kinds: params.kinds,
-          startKey: params.startKey,
+          startKey: params.startKey || '',
           search: params.search,
           query: params.query,
           pinnedOnly: params.pinnedOnly,
           searchAsRoles: params.searchAsRoles,
-        });
-        if (params.sortBy) {
-          req.sortBy = api.SortBy.create({
-            field: params.sortBy.field,
-            isDesc: params.sortBy.isDesc,
-          });
+          sortBy: params.sortBy || { field: 'name', isDesc: false },
```

Comment:
Where are the defaults coming from? I have a feeling they should be defined one layer up.

Reply:
Hmm the `sortBy` default is not needed, it's a leftover from testing.
I fixed the default value for `startKey`, it should always be a string.